File: ExampleSolutions/ExampleClassLibraryVb/PasswordHasher.vb
Option Explicit On

Imports System.Linq
Imports System.Security.Cryptography
Imports System.Text

Public Class PasswordHasher
    Public Shared Function HashPassword(password As String) As String
        Return SHA256.Create().ComputeHash(Encoding.UTF8.GetBytes(password)).Aggregate(New StringBuilder(), Function(sb, b) sb.AppendFormat("{0:x2}", b)).ToString()
    End Function
End Class

File: ExampleSolutions/ExampleClassLibrary/PasswordHasher.cs
using System.Linq;
using System.Security.Cryptography;
using System.Text;

namespace ExampleClassLibrary
{
    internal static class PasswordHasher
    {
        internal static string HashPassword(string password)
        {
            return SHA256.Create().ComputeHash(Encoding.UTF8.GetBytes(password)).Aggregate(new StringBuilder(), (sb, b) => sb.AppendFormat("{0:x2}", b)).ToString();
        }
    }
}
